### PR TITLE
version 2.1.0 changes added

### PR DESCRIPTION
**Pull request by cs-pranali-chavan**
- API call updated for every module selected in each layer
- Widget name updated
- Minor bug fixes
- Null values excluded while rendering Sankey chart
- Support added for text fields in target nodes selection

### CODE REVIEW

**Comment by cs-pranali-chavan** [comment]
Will merge this after concluding with Abhishek sir about the new changes and test with some use cases

File: widget/edit.controller.js [hunk 1]
```diff
@@ -142,20 +153,21 @@
   
       //Check if target is picklist or manyToMany
       function checkTargetType(_index) {
-        let isPicklist = _.filter($scope.params.targetNodeFields, function (field) {
+        let _currentTargetData = _.filter($scope.params.targetNodeFields, function (field) {
           return field.name === $scope.config.layers[_index].targetNodeField
         });
-        if (isPicklist.length > 0 && isPicklist[0]['type'] === 'manyToMany') {
-          var targetEntity = new Entity($scope.config.layers[_index].targetNodeField);
+        if (_currentTargetData && _currentTargetData.length > 0 && (_currentTargetData[0]['type'] === 'manyToMany' || _currentTargetData[0]['type'] === 'picklist')) {
```
Comment (cs-vikas-shelar) [inline widget/edit.controller.js]: '_currentTargetData[0]['type'] === 'picklist'' check is not required here.

File: widget/info.json [hunk 2]
```diff
@@ -12,9 +12,9 @@
...
         ],
         "snapshots": [
-            "https://repo.fortisoar.fortinet.com/fsr-widgets/socOverviewSankey-2.0.0/images/soc-sankey-edit-live-data.png",
-            "https://repo.fortisoar.fortinet.com/fsr-widgets/socOverviewSankey-2.0.0/images/soc-sankey-edit-static-data.png",
-			"https://repo.fortisoar.fortinet.com/fsr-widgets/socOverviewSankey-2.0.0/images/soc-sankey-view.png"
+            "https://repo.fortisoar.fortinet.com/fsr-widgets/socOverviewSankey-2.1.0/images/soc-sankey-edit-live-data.png",
+            "https://repo.fortisoar.fortinet.com/fsr-widgets/socOverviewSankey-2.1.0/images/soc-sankey-edit-static-data.png",
+			"https://repo.fortisoar.fortinet.com/fsr-widgets/socOverviewSankey-2.1.0/images/soc-sankey-view.png"
         ],
         "help_online":"https://github.com/fortinet-fortisoar/widget-soc-overview-sankey/blob/release/2.0.0/README.md",
```
Comment (cs-vikas-shelar) [inline widget/info.json]: Need to correct the version in the help path.

File: widget/info.json [hunk 3]
```diff
@@ -1,8 +1,8 @@
 {
     "name": "socOverviewSankey",
-    "title": "SOC Overview Sankey",
+    "title": "Sankey",
     "subTitle": "Visualise flow between nodes in a directed acyclic network.",
-    "version": "2.0.0",
+    "version": "2.1.0",
     "published_date": 1711648363,
     "releaseNotes": "unavailable",
```
Comment (cs-vikas-shelar) [inline widget/info.json]: Please add release notes and change status here as available.

File: widget/release_notes.md [hunk 4]
```diff
@@ -3,4 +3,10 @@
 - In *Edit View* of the SOC Overview Sankey Chart has been completely revamped to add a **Data Source** field using which users can choose whether to render static or live data:
     - **Record Containing JSON Data** - To render all records in a single keystore record that contains a JSON object. 
     - **Get Live Data** - To dynamically render nodes and links data. The live data is queried based on the selected module and fields.
-- Added internationalization support.
\ No newline at end of file
+- Added internationalization support.
```
Comment (cs-vikas-shelar) [inline widget/release_notes.md:2]: May be here need to clear all previous release notes added. Only new content will remain in the file.

File: widget/widgetAssets/js/socOverviewSankey.service.js [hunk 5]
```diff
@@ -56,44 +58,91 @@
...
+                              _resource = previousLayer['targetNodeModule'];
+                          }
+                          let _fieldCondition = getSubTargetFieldCondition(_resource,previousLayer);
+                          //if targetSubfield is not null then use targetNodeSubfield 
+                          //else use target node
+                          if (previousLayer['targetNodeSubField'] !== null) {
+                              queryObject.aggregates.push({
+                                  'operator': 'groupby',
+                                  'alias': 'series_' + elementIndex,
+                                  'field': previousLayer['targetNodeSubType'] === 'picklist' || previousLayer['targetNodeSubType'] === 'manyToMany' ? previousLayer['targetNodeSubField'] + '.itemValue' : previousLayer['targetNodeSubField']
```
Comment (cs-vikas-shelar) [inline widget/widgetAssets/js/socOverviewSankey.service.js]: previousLayer['targetNodeSubType'] === 'manyToMany' : Not required here.

File: widget/widgetAssets/js/socOverviewSankey.service.js [hunk 6]
```diff
@@ -102,66 +151,72 @@
...
+                      'alias': 'color_series_' + elementIndex,
+                      'field': currentLayer['targetNodeField'] + '.' + currentLayer['targetNodeSubField'] + '.color'
+                  });
+              }
+          }
   
+          //the condition of field is updated if the resource selected on each layer is different or similar
+          function getSubTargetFieldCondition(_resource, currentLayer) {
+              let _fieldCondition = currentLayer['targetNodeSubField'];
+              if(currentLayer['targetNodeSubType'] === 'picklist' || currentLayer['targetNodeSubType'] === 'manyToMany')
```
Comment (cs-vikas-shelar) [inline widget/widgetAssets/js/socOverviewSankey.service.js]: currentLayer['targetNodeSubType'] === 'manyToMany' check is not required.

**Review by cs-vikas-shelar — COMMENTED** [review]
Code changes look fine. Few minor corrections required, added minor comments.
Please add support for default static data selection, which will be used in the Solution Pack for default dashboard creation.

File: widget/edit.controller.js (hunk 1 above)
Comment (cs-pranali-chavan) [reply widget/edit.controller.js]: removed

File: widget/info.json (hunk 2 above)
Comment (cs-pranali-chavan) [reply widget/info.json]: updated

File: widget/info.json (hunk 3 above)
Comment (cs-pranali-chavan) [reply widget/info.json]: updated

File: widget/release_notes.md [hunk 7]
```diff
@@ -3,4 +3,10 @@
...
+- API call updated for every module selected in each layer
```
Comment (cs-suyog-jinturkar) [inline widget/release_notes.md]: Release note needs to be less technical and more user friendly

File: widget/view.controller.js [hunk 8]
```diff
@@ -2,486 +2,613 @@
...
+        id: 'btn-6m',
+        text: 'BUTTON_LAST_6_MONTHS',
+        onClick: function () {
+          refreshSankey(90, 'btn-6m');
+        },
+        active: true,
+        type: 'submit'
+      }, {
+        id: 'btn-3m',
+        text: 'BUTTON_LAST_3_MONTHS',
```
Comment (cs-yoges-chauriwar) [inline widget/view.controller.js:42]: use `$scope.viewWidgetVars.BUTTON_LAST_3_MONTHS` to access button text for all buttons

**Comment by cs-yoges-chauriwar** [comment]
Looks Good

**Pull request merged**